src/toil/wdl/toilwdl.py

```diff
@@ -0,0 +1,2077 @@
+# Copyright (C) 2018 UCSC Computational Genomics Lab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+import fnmatch
+import argparse
+import json
+import os
+import collections
+import subprocess
+import toil.wdl.wdl_parser as wdl_parser
+import sys
+import json
+import logging
+
```

Comment:
A paragraph describing the general design would be useful here.

Reply:
Moved the class up and the two orphaned defs down so that it's more visible.  Hope it's enough detail for a general overview.